app/views/listings/index.html.erb

```diff
@@ -63,14 +63,35 @@
                 <p class="text-gray-800 font-bold"><%= number_to_currency(listing.price) %></p>
               </div>
               <div class="flex flex-row">
-                <p><%= listing.course_number %></p>
-                <p class="ml-4"># <%= listing.isbn %></p>
-                <p class="ml-4"><%= listing.listing_type %></p>
+                <% if listing.course_number.present? %>
+                  <p><%= listing.course_number %></p>
+                <% end %>
+                <% if listing.edition.present? %>
+                  <p class="ml-4"><%= listing.edition %></p>
+                <% end %>
+                <% if listing.isbn.present? %>
+                  <p class="ml-4"># <%= listing.isbn %></p>
+                <% end %>
+                <% if listing.listing_type.present? %>
+                  <p class="ml-4"><%= listing.listing_type %></p>
+                <% end %>
               </div>
             </div>
-            <div class="flex flex-row justify-between">
-              <p>A</p>
-              <p>B</p>
+            <div class="flex flex-col items-end justify-between">
+              <% saved_item = current_user.saved_items.find_by(listing_id: listing.id) %>
+              <% if saved_item.present? %>
```

Comment:
This button for managing the saved state of a listing would be a good thing to move to a partial as long as your styling for it is consistent. That way anywhere you want to add the button to save a listing you can just use the partial.

I think this would be an even better fit for a [view_component](https://viewcomponent.org/) if you want to add this gem. We use it on pretty much all of our projects.